### PR TITLE
`azurerm_windows_web_app` - fix `health_check_eviction_time_in_min` not being set issue

### PR DESCRIPTION
**Pull request by xiaxyi**
This pr is fixing two issues:
1. The property `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` in app_setting is suppose to be set if user assign a value to the property `health_check_eviction_time_in_min`.
2. The `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` should be written back to the state file if user explicitly specify it in config.

fix issues:https://github.com/hashicorp/terraform-provider-azurerm/issues/17123
https://github.com/hashicorp/terraform-provider-azurerm/issues/17546
https://github.com/hashicorp/terraform-provider-azurerm/issues/16302

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @xiaxyi - Thanks for this PR. I've left a couple comments below for changes I believe to be needed. But otherwise I think it looks good 👍 

Thanks!

File: internal/services/appservice/helpers/web_app_schema.go [hunk 1]
```diff
@@ -3785,8 +3785,8 @@ func ExpandAppSettingsForCreate(settings map[string]string) *[]web.NameValuePair
 	return nil
 }
 
-func FlattenAppSettings(input web.StringDictionary) (map[string]string, *int) {
-	maxPingFailures := "WEBSITE_HEALTHCHECK_MAXPINGFAILURE"
+func FlattenAppSettings(input web.StringDictionary, d sdk.ResourceMetaData) (map[string]string, *int) {
+	maxPingFailures := "WEBSITE_HEALTHCHECK_MAXPINGFAILURES"
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/appservice/helpers/web_app_schema.go:3789]: I think we need to move the `WEBSITE_HEALTHCHECK_MAXPINGFAILURES` from the `unmanagedSettings` list if user adds the key explicitly in the config?

File: internal/services/appservice/helpers/web_app_schema.go [hunk 2]
```diff
@@ -3803,6 +3803,12 @@ func FlattenAppSettings(input web.StringDictionary) (map[string]string, *int) {
 		healthCheckCount = &h
 	}
 
+	if userInputAppSetting := d.ResourceData.Get("app_settings").(map[string]interface{}); userInputAppSetting != nil {
+		if _, ok := userInputAppSetting[maxPingFailures]; ok {
+			unmanagedSettings = unmanagedSettings[:len(unmanagedSettings)-1]
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (xiaxyi) [reply internal/services/appservice/helpers/web_app_schema.go:3811]: But azure provided such way to set this property and lots of users are aware of this usage as well. shall we just let them set it via app_setting?

**Comment by xiaxyi** [comment]
Thank @jackofallops for the comments, left some feedbacks inline

File: internal/services/appservice/helpers/web_app_schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (katbyte) [reply internal/services/appservice/helpers/web_app_schema.go:3811]: we have moved it out of the map into its own property, therefor it should only be set there as @jackofallops, and if the user does put it into the app settings map we should maybe toss up an error telling the user to set it there?

**Comment by katbyte** [comment]
also we have some test failureS:
image

**Comment by xiaxyi** [comment]
Thanks @jackofallops for the comments. I removed the app_setting KVP code change, also, the update check of app_setting should be removed as we are collecting data from site_config, which means, app_setting still needs to be updated even there is no change in the config.

For the test cases, I removed the property checks as we have the import step.

**Comment by xiaxyi** [comment]
> we still have test failures image

Thanks @katbyte for the comment, the acc test failure will be fixed in https://github.com/hashicorp/terraform-provider-azurerm/pull/18177

File: internal/services/appservice/helpers/web_app_schema.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmoetzing) [reply internal/services/appservice/helpers/web_app_schema.go]: @katbyte If think the error is really important. I fixed this issue by placing the env variable in `app_settings` and probably many others will have too.

**Comment by pierluca** [comment]
Is there any ETA on when this issue will be fixed/merged/released ?

**Comment by jackofallops** [comment]
Closing as superseded by 19685

**Pull request closed without merging**